app/scripts/components/common/dateslider/index.tsx

```diff
@@ -70,9 +74,28 @@ export default function DateSliderControl(props: DateSliderControlProps) {
     }
   }, []);
 
+  // Limit the data that is rendered so that performance is not hindered by very
+  // large datasets.
+  // Lower and Upper pixel bounds of the data that should be rendered taking the
+  // horizontal drag window into account.
+  const minX = zoomXTranslation * -1;
+  const maxX = width - zoomXTranslation;
+  // Using the scale, get the indexes of the data that would be rendered.
+  const indexes = [
+    Math.max(Math.floor(x.invert(minX)), 0),
+    // Add one to account for when there's a single data point.
+    Math.ceil(x.invert(maxX)) + 1
+  ];
+
+  const dataToRender = useMemo(
+    () => data.slice(...indexes),
+    [data, ...indexes]
+  );
```

Comment:
What are you trying to achieve with that `useMemo`? 
From what I understand from the whole block, this could be wrapped in a bigger useMemo with a dependency on data and zoomXTranslation. 

&lt;pedantic&gt;`indices` rather than `indexes`&lt;/pedantic&gt;

Reply:
`zoomXTranslation` translation has a subpixel value, so the useEffect would run constantly.
The indexes are more stable (i.e. the `zoomXTranslation` would have to change "a lot" before having an index change).

Will update the name ;)